internal/ts_library/rule.bzl

```diff
@@ -1,286 +1,102 @@
+        return path[:-4] + ".js"
+    elif path.endswith(".ts"):
+        return path[:-3] + ".js"
+    else:
+        return path
 
 ts_library = rule(
-  implementation=_ts_library_impl,
-  attrs = {
-    "srcs": attr.label_list(
-      allow_files = True,
-      mandatory = True,
-    ),
-    "deps": attr.label_list(
-      providers = [
-        [JsLibraryInfo],
-        [NpmPackagesInfo],
-      ],
-      default = [],
-    ),
-    "tsconfig": attr.label(
-      allow_files = [".json"],
-      single_file = True,
-      default = Label("//internal/ts_library:default_tsconfig.json"),
-    ),
-    "_internal_nodejs": attr.label(
-      allow_files = True,
-      single_file = True,
-      default = Label("@nodejs//:node"),
-    ),
-    "_internal_packages": attr.label(
-      default = Label("//internal:packages"),
-    ),
-    "_ts_library_create_full_src_script": attr.label(
-      allow_files = True,
-      single_file = True,
-      default = Label("//internal/ts_library:create_full_src.js"),
-    ),
-    "_ts_library_compile_script": attr.label(
-      allow_files = True,
-      single_file = True,
-      default = Label("//internal/ts_library:compile.js"),
-    ),
-    "_ts_library_transpile_script": attr.label(
-      allow_files = True,
-      single_file = True,
-      default = Label("//internal/ts_library:transpile.js"),
-    ),
-    "_empty_npm_packages": attr.label(
-      default = Label("//internal/npm_packages/empty:packages"),
-    ),
-  },
-  outputs = {
-    "compilation_src_dir": "%{name}_compilation_src",
-    "compiled_dir": "%{name}_compiled",
-    "transpilation_src_dir": "%{name}_transpilation_src",
-    "transpiled_dir": "%{name}_transpiled",
-  },
+    implementation = _ts_library_impl,
+    attrs = dict(JS_LIBRARY_ATTRIBUTES, **{
+        "tsconfig": attr.label(
+            allow_files = [".json"],
+            single_file = True,
+            default = Label("//internal/ts_library:default_tsconfig.json"),
+        ),
+        "_ts_library_create_full_src_script": attr.label(
+            allow_files = True,
+            single_file = True,
+            default = Label("//internal/ts_library:create_full_src.js"),
+        ),
+        "_ts_library_compile_script": attr.label(
+            allow_files = True,
+            single_file = True,
+            default = Label("//internal/ts_library:compile.js"),
+        ),
+        "_ts_library_transpile_script": attr.label(
+            allow_files = True,
+            single_file = True,
+            default = Label("//internal/ts_library:transpile.js"),
+        ),
+        "_ts_config_genscript": attr.label(
+            allow_files = True,
+            single_file = True,
+            default = Label("//internal/ts_library:tsconfig.gen.js"),
+        )
+    }),
+    outputs = {
+        "compilation_src_dir": "%{name}_compilation_src",
+        "compiled_dir": "%{name}_compiled",
+        # "transpilation_src_dir": "%{name}_transpilation_src",
```

Comment:
Is this no longer needed?

Reply:
I don't think so, but I know that you added it in the first place because using TypeScripts fast transpilation created a shorter critical path. Not sure if there are other rules using these outputs and looks like I might find out once I get to fixing the build for this PR.